### ISSUE TITLE
Make Font-Validator the submodule?

### ISSUE THREAD

**Issue by threeseat** [HinTak/FontVal-MacGUI#2]
This may be water under the bridge, but...

The main Font-Validator project and its products—executables for Windows, Linux, and MacOS, plus GUIs for Windows and Linux—are all independent of the Mac GUI.

But the Mac GUI depends on the command-line executable and the help-file information.  (I should soon have a pull request for a means to keep the Mac help up-to-date with OurData.xml.)

Making Font-Validator a submodule of FontVal-MacGUI would better reflect those dependencies.  If made a submodule, Font-Validator could be self-contained (assuming the FreeType library is added back to the main project).  The mkbundle'd Mac binary would exist only in one place (and still in a subfolder, allowing for easy Mac GUI access).  Similarly, the Mac GUI could pull the help-file information from a subfolder.  And any changes to the derived product wouldn't need to be pushed upstream.

The restructuring would require changes within the Xcode project, but primarily just the file references.

If you think this would make sense, I can work on creating a pull request.  Unless @schriftgestalt , you would prefer to do that.

**Comment by schriftgestalt** [comment]
That sounds like a good idea. I don't have time to do that right now.

**Comment by HinTak** [comment]
The html help files are made by the "make gendoc" target's tool. Though @schriftgestalt made some hand editing one-off to fit the mac bundle . HTH. Automating what is generated to fit what mac bundle needs is a good idea. I startes on a bundle of perl regex's a while ago but didn't finish. If you want I can try finding it ( assuming you want to use perl...) - but probably best just start new.

To make a new command-line binary, you need to pull in
https://github.com/HinTak/mono-modification too (to work around the code-signing issue) , plus a version-matching mono installation.

The libfreetype is moved (copied, rather) ing MacGUI - but still exists in Font-Validator. I think some playing with `install_name_tool` is needed to get the binary to load a libfreetype on its side. At the moment I don't plan to put it out in source form, but I see there will be two ways of running mkbundle, one to hide libfreetype completely as a standalone command-line binary (like it currently is), another to leaving it outside, with some help from `install_name_tool`, to allow both parts to be code-signed in a mac bundle.

Basically I don't mind it either way, considered that the overlap is quite small. I am mostly against organizational changes like this like which should be the submodule, mostly because it is largely a personal preference (neither option is "better") and does not add value to the outcome.
Practically, it is just modifying two `.gitmodules` files, (both, to avoid recursion) so it is quite painless, if there is a genuine reason for doing it. The fact is, very few people would actually be doing `git submodule update` in either form.

**Comment by HinTak** [comment]
About the html help files - about a dozen are in "NewHelp" directory and not auto-generated from OurData.xml - that's a historical issue as they were not available in source form somehow (or rather, had not been converted to inside OurData) and so where recovered from the last public release from Microsoft. There is one practical issue from this: they are not referenced from the generated table-of-content. I have a small patch file which patches the generated toc.

So the steps of making the Windows chm file are:

- `make gendoc` to build the tool

- follow the message at the end of above, to generate the html's

- patch the toc

- run Microsoft 's chm compiler

The mac help files were a one-off edit of the stage just before the last.
